src/components/connect-request.js

```diff
@@ -27,62 +45,69 @@ const connectRequest = (mapPropsToConfig, options = {}) => (WrappedComponent) =>
         }
 
         componentDidMount() {
-            this.requestAsync(this.props, false, true);
+            const configs = mapPropsToConfig(this.props);
+            this.requestAsync(configs, false, true);
         }
 
         componentDidUpdate(prevProps) {
-            const prevConfig = mapPropsToConfig(prevProps);
-            const config = mapPropsToConfig(this.props);
-            const prevQueryKey = prevConfig && getQueryKey(prevConfig.url, prevConfig.body);
-            const queryKey = config && getQueryKey(config.url, config.body);
-
-            if (prevQueryKey !== queryKey) {
-                this.cancelPendingRequests();
-                this.requestAsync(this.props, false, true);
+            const prevConfigs = toArray(mapPropsToConfig(prevProps)).filter(Boolean);
+            const configs = toArray(mapPropsToConfig(this.props)).filter(Boolean);
+
+            const { cancelKeys, requestKeys } = diffConfigs(prevConfigs, configs);
+            const requestConfigs = prevConfigs.filter((c) => {
+                return requestKeys.includes(getQueryKey(c.url, c.body));
+            });
+
+            if (cancelKeys.length) {
+                this.cancelPendingRequests(cancelKeys);
```

Comment:
We should filter/check that all of these keys exist in `this._pendingRequests` or else warnings will get logged from query middleware.